cypress/e2e/bookStore.cy.js

```diff
@@ -1,11 +1,80 @@
-/// <reference types='cypress' />
+/// <reference types="cypress" />
+/// <reference types="..support'" />
 
 describe('Book Store app', () => {
-  before(() => {
-    
+  const testData = {
+    user: {
+      username: 'Anton3108',
+      password: 'A12345678a!'
+    },
+    book: {
+      name: 'Speaking JavaScript',
+      author: 'Axel Rauschmayer',
+      publisher: 'O\'Reilly Media',
+      description: 'Like it or not, JavaScript is everywhere'
+    }
+  };
+
+  beforeEach(() => {
+  cy.visit('https://demoqa.com/login');  
   });
 
-  it('', () => {
-    
+  it('should provide an ability to login', () => {
+  cy.get('#userName')
+  .type(testData.user.username);
+
+  cy.get('#password')
+  .type(testData.user.password);  
+
+  cy.get('#login')
+  .click();
+
+  cy.get('#userName-value')
+  .should('contain', testData.user.username);
+  
+  cy.url()
+  .should('include', '/profile');
   });
+
+it('should provide an ability to search for the book and add to cart', () => {
+cy.login(testData.user.username, testData.user.password);
+
+//cy.visit('https://demoqa.com/profile');
+
+cy.contains('#item-2', 'Book Store')
+.click();
+
+cy.get('#searchBox')
+.type(testData.book.name);
+
+cy.contains('a', testData.book.name)
+.click();
+
+cy.contains('.form-label', 'Like it or not, JavaScript is everywhere these days-from browser to server to mobile-and now you, too, need to learn the language or dive deeper than you have. This concise book guides you into and through JavaScript, written by a veteran programmer who o')
```

Comment:
move the text to a variable

Reply:
Not fixed. Create the variable for storing this text.